packages/terra-form-validation/docs/MaxLengthInput.md

```diff
@@ -0,0 +1,232 @@
+
+```diff
+render() {
+  return (
+    <Spacer marginBottom="small">
+      <Form
+        onSubmit={this.submitForm}
+        render={this.renderForm}
+      />
++     {this.state.submittedValues
++       && (
++       <div>
++         <p>Form Submitted Successfully With</p>
++         <pre>{JSON.stringify(this.state.submittedValues, 0, 2)}</pre>
++       </div>
++       )
++     }
+    </Spacer>
+  );
+}
+```
+
+Submit will now visually update with a JSON of our `submittedValues` state.
+
+We don't need a specific initial value but let's set an initial value that is an empty string so we have an easy place to redefine it if need be. We add an object to our `<Form>`. We define our object with the property being the inputID of our `<InputField>`, which in this case is `example`, and the property's value being the value of the `<InputField>`.
+```diff
+render() {
+  return (
+    <Spacer marginBottom="small">
+      <Form
+        onSubmit={this.submitForm}
+        render={this.renderForm}
++       initialValues={{ example: '' }}
+      />
+      ...
+    </Spacer>
+  );
+}
+```
+
+Finally we need to validate our input field. This has two parts to it, a `validate` prop on our `<Form>` component and a `validate` prop on our `<Field>` component in our `renderForm` function.
+
+The `<Form>` component's `validate` prop takes a validation function that reads all form values and returns validation errors. Let's create a function that checks the existence of `example` field's values. On error, set the ID to `required`. This will be displayed if validation fails.
+
+```diff
+render() {
+  return (
+    <Spacer marginBottom="small">
+      <Form
+        onSubmit={this.submitForm}
+        render={this.renderForm}
+        initialValues={{ example: '' }}
++       validate={(values) => {
++           const errors = {};
++           if (!values.example) {
++             errors.example = 'Required';
++           }
++           return errors;
++       }}
+      />
+      ...
+    </Spacer>
+  );
+}
+```
+
+Our `validate` function returns an object that has properties, with the property being our ID, and its value being a string we want displayed for when it fails to validate.
+
+The `<Field>` component's `validate` prop is similar to `<Form>`'s, with a main caveat; it only validate's it's value instead of all the values. Let's create a function that checks if the length is greater than 7. This function takes a single argument (value of our `<Field>`) and returns a string if this criteria is met.
+
+```javascript
+const validateLength = (name) => {
+  if (name && name.length > 7) {
+    return 'Name needs to be less than 8 characters long';
+  }
+
+  return undefined;
+};
+```
+
+Finally, let's add pass the function as the `validate` prop to `<Field>` within our `renderForm` function..
```

Comment:
nit: Let's go with `Finally, let's add the function as the` and remove the double period at the end of the sentence.

Reply:
Fixed here https://github.com/cerner/terra-framework/pull/635/commits/415c46ab9a0b74893a686c38430b1d8decd708ad#diff-00d3067043a9948e4371f570ef81a357R201